### DISCUSSION TITLE
How to work with the challenge QR code after every lesson?

### DISCUSSION THREAD

**Discussion by KaiYueWong** [Cyfrin/foundry-full-course-cu#562] [Q&A] [answered]
As described in the title, is there any documentation about the challenge part after each lesson?
I just saw a QR code to etherscan after the lesson.
Many thanks.

**Comment by shikhar229169** [comment]
After going to the challenge contract on etherscan, click on contract over there you will find Lesson One contract and you need to call the solveChallenge function for each lesson. Above which there is a comment which tells you about the parameters and what you have to do.

**Comment by KaiYueWong** [comment]
Thanks for answering. I found the solveChallenge function. Where is the comment about what I have to do and what about the renounceOwnership and transferOwnership? I am still trying to understand how should I read the challenge task.
![testnetImage](https://github.com/Cyfrin/foundry-full-course-f23/assets/22658105/ccee26e2-c619-4fa9-ac53-27d6174e5e9f)

**Reply by shikhar229169** [reply]
Go to the code section, there you will find the contract.

You just have to focus on the solveChallenge function.
Just ignore other functions for now you will get the idea about them in the course later.

**Comment by KaiYueWong** [comment]
Thanks for your reply. I noticed that we can use Arbitrum or Sepolia for doing the challenge. 

For lesson 3, I cannot send the transaction. The error is :
![testnetImage](https://github.com/Cyfrin/foundry-full-course-f23/assets/22658105/c28683c2-32eb-42eb-ae88-16d48bc62ad6)

**Reply by cromewar** [reply, marked as answer]
Hint: don't forget to first read the contract to find what the position actually is 😉
